internal/api/token_oidc.go

```diff
@@ -201,7 +201,7 @@ func (a *API) IdTokenGrant(ctx context.Context, w http.ResponseWriter, r *http.R
 
 		user, terr = a.createAccountFromExternalIdentity(tx, r, userData, providerType)
 		if terr != nil {
-			if errors.Is(terr, errReturnNil) {
+			if errors.Is(terr, errEmailVerificationRequired) {
```

Comment:
Can you please double check this, as `errReturnNil` was a special case used elsewhere. Maybe best to just remove it completely and see what does not compile.

Reply:
yeah i've double checked this - we used to return `errReturnNil` as a hack to when we should really be returning a proper error type and checking it 

this is how it's handled before: 
[1]: https://github.com/supabase/gotrue/blob/d76f439b65413803ccf37cf8a217a932addfb477/internal/api/external.go#L199
[2]: if `errReturnNil`, then we assume that the token and flowState will be nil and handle the error [here](https://github.com/supabase/gotrue/blob/d76f439b65413803ccf37cf8a217a932addfb477/internal/api/external.go#L247) which is super confusing

Reply:
oh yeah my bad, i just realised we used to return `errReturnNil` so that the main transaction won't be rolled back - will revert this change and add a comment to make it clear